### PR TITLE
feat: [CDS-56445]: overrides 2.0 project level framework changes

### PR DESCRIPTION
**Pull request by AshwinBhatkal-Harness**
### Summary

Overrides 2.0 project level framework changes

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/10-common/RouteDefinitions.ts [hunk 1]
```diff
@@ -979,6 +979,24 @@ const routes = {
       return queryString.length > 0 ? `${routePath}?${queryString}` : routePath
     }
   ),
+  toServiceOverrides: withAccountId(
+    ({
+      orgIdentifier,
+      projectIdentifier,
+      module,
+      accountRoutePlacement
```
Comment (pankajsingh-harness) [inline src/modules/10-common/RouteDefinitions.ts:987]: Are we not supporting query params for this route?

File: src/modules/75-cd/components/ServiceOverrides/Environment/GlobalOverrides.tsx [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import React from 'react'
+
+import { Container } from '@harness/uicore'
+
+export default function GlobalEnvironmentOverrides(): React.ReactElement {
+  return <Container padding={{ right: 'xlarge', left: 'xlarge' }}>Global Environment Overrides</Container>
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/ServiceOverrides/Environment/GlobalOverrides.tsx:6]: This label should comes from yaml

File: src/modules/75-cd/components/ServiceOverrides/Environment/ServiceSpecific.tsx [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import React from 'react'
+
+import { Container } from '@harness/uicore'
+
+export default function EnvironmentServiceSpecificOverrides(): React.ReactElement {
+  return <Container padding={{ right: 'xlarge', left: 'xlarge' }}>Environment Service Specific Overrides</Container>
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/ServiceOverrides/Environment/ServiceSpecific.tsx:6]: Same as above

File: src/modules/75-cd/components/ServiceOverrides/Infrastructure/GlobalOverrides.tsx [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import React from 'react'
+
+import { Container } from '@harness/uicore'
+
+export default function GlobalInfrastructureOverrides(): React.ReactElement {
+  return <Container padding={{ right: 'xlarge', left: 'xlarge' }}>Global Infrastructure Overrides</Container>
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/ServiceOverrides/Infrastructure/GlobalOverrides.tsx:6]: Same as above

File: src/modules/75-cd/components/ServiceOverrides/Infrastructure/ServiceSpecific.tsx [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import React from 'react'
+
+import { Container } from '@harness/uicore'
+
+export default function InfrastructureServiceSpecificOverrides(): React.ReactElement {
+  return <Container padding={{ right: 'xlarge', left: 'xlarge' }}>Infrastructure Service Specific Overrides</Container>
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/ServiceOverrides/Infrastructure/ServiceSpecific.tsx:6]: Same as above

File: src/modules/75-cd/components/ServiceOverrides/Environment/GlobalOverrides.tsx (hunk 2 above)
Comment (AshwinBhatkal-Harness) [reply src/modules/75-cd/components/ServiceOverrides/Environment/GlobalOverrides.tsx:6]: This is temporary changes. Behind FF. Need to have a new component here. Will do it in next PR

File: src/modules/75-cd/components/ServiceOverrides/Environment/ServiceSpecific.tsx (hunk 3 above)
Comment (AshwinBhatkal-Harness) [reply src/modules/75-cd/components/ServiceOverrides/Environment/ServiceSpecific.tsx:6]: Same as above

File: src/modules/75-cd/components/ServiceOverrides/Infrastructure/GlobalOverrides.tsx (hunk 4 above)
Comment (AshwinBhatkal-Harness) [reply src/modules/75-cd/components/ServiceOverrides/Infrastructure/GlobalOverrides.tsx:6]: Same as above

File: src/modules/75-cd/components/ServiceOverrides/Infrastructure/ServiceSpecific.tsx (hunk 5 above)
Comment (AshwinBhatkal-Harness) [reply src/modules/75-cd/components/ServiceOverrides/Infrastructure/ServiceSpecific.tsx:6]: Same as above

**Review by pankajsingh-harness — APPROVED** [review]

File: src/modules/10-common/RouteDefinitions.ts (hunk 1 above)
Comment (AshwinBhatkal-Harness) [reply src/modules/10-common/RouteDefinitions.ts:987]: Not yet. Will do it in next set of changes

**Review by bhavya-sinha — APPROVED** [review]

**Review by vikrant-harness — APPROVED** [review]

**Review by bhavya-sinha — APPROVED** [review]

**Comment by AshwinBhatkal-Harness** [comment]
Coverage for this PR will be done as part of https://harness.atlassian.net/browse/CDS-69134

**Pull request merged**